Adapp 18865 rework rs api docs

Sporiff opened the pull request:
## Related issues

<!-- What issue does this pull request resolve? Add a link to a Jira or GitHub issue. -->

- Jira issue link: https://adjustcom.atlassian.net/browse/ADAPP-18865

## Changes

This PR reworks the current report service documentation into a single guide.

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+### Retrieve filters {% #retrieve-filters %}
+
+When querying the Reports Service API, you need to filter your reporting data. The Reports Service API supports different filters such as metrics and dimensions. A full list of metrics is available in the [Datascape metrics glossary](https://help.adjust.com/en/article/datascape-metrics-glossary) and a full list of dimensions is available in the [Datascape dimensions glossary](https://help.adjust. ...
```
I'd discourage people from using `filters_data` for getting metrics, while useful in some cases, filters_data don't contain all metrics and are there mostly just for historical reasons.

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+| `partners`                                                                                              | Returns a list of partners.                                                                             |
+| `ad_revenue_sources`                                                                                    | Returns a list of ad revenue sources.                                                                   |
+| `iap_revenue_mode`                                                                                      | Returns a list of ad revenue modes.                                                                     |
+| `attribution_type`                                                                                      | Returns a list of attribution types. Examples: all, click, impression.                                  |
```
should there be some explanation about the meaning of the filter itself? I think we have that elsewhere but I don't see it linked anywhere

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+`metrics`: `String`
+
+: A comma-separated list of metrics to report on. You can fetch metrics from the [`filters_data` endpoint](#retrieve-filters) or find a full list in the [Datascape metrics glossary](https://help.adjust.com/en/article/datascape-metrics-glossary).
```
see my comment above about using filters_data for metrics

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+| `ad_spend_mode`                                                                                         | Returns a list of ad spend modes. Examples: attribution, network, mixed.                                |
+| `utc_offset`                                                                                            | The offset (in hours) from UTC. Defaults to 0 (UTC).                                                    |
+| `attributes`                                                                                            | Returns a list of attributes.                                                                           |
+| `index`                                                                                                 | Returns a list of key metrics that relate to all apps.                                                  |
```
index returns dimensions, not metrics

SlavaSkvortsov replied:
Let's just remove it?

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+The following filters are available:
+
+| Filter                                                                                                  | Description                                                                                             |
```
There are filters missing from this list:
```
- channels
- platform
- sandbox
```

jmikovic commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+`attribution_type`: `String`
+
+: The type of engagement the attribution awards.
```
`attribution_type` can be confusing to some users, I'd recommend referring to this [article](https://help.adjust.com/en/article/set-up-your-view-in-datascape?src=search_page#attribution-type) that explains how it changes the data

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+: A substring to filter a given dimension by suffix. Case insensitive.
+
+`[metric]__lt`: `String`
+
+: "Less than" filter. Returns only values lower than the specified value for the specified metric.
+
+`[metric]__lte`: `String`
+
+: "Less than or equal to" filter. Returns only values lower than or equal to the specified value for the specified metric.
+
+`[metric]__gt`: `String`
+
+: "Greater than" filter. Returns only values higher than the specified value for the specified metric.
+
+`[metric]__gte`: `String`
+
+: "Greater than or equal to" filter. Returns only values greater than or equal to the specified value for the specified metric.
+
+`[metric]__eq`: `String`
+
+: "Equal to" filter. Returns only values equal to the specified value for the specified metric.
+
+`[metric]__ne`: `String`
+
+: "Not equal to" filter. Returns only values not equal to the specified value for the specified metric.
+{% /deflist %}
```
All metrics are numerical values, so all these filter values must be numerical as well, not just `String`

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
````diff
@@ -0,0 +1,705 @@
+}
+```
+
+### Fetch a pivot report {% #fetch-pivot-report %}
````
For the last month, only 0.34% of API requests made by users were for the pivot endpoint
<img width="521" alt="image" src="https://github.com/user-attachments/assets/7fb781cd-faae-4269-9f2b-6a9912fd5cfc">

I'd remove pivot from the documentation completely, it's not something customers actually need, this endpoint was mostly created to empower our frontend

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+{% deflist %}
+`readable_names`: `Boolean`
+
+: Whether to return columns with their human-readable names. Defaults to `false`.
```
We strongly recommend all users to provide `true` for this param. We were even considering changing the default, but it has too much of an impact. Let's emphasize that we think that it's beneficial for everyone to use it

SlavaSkvortsov replied:
I'd also added it to the example URL and checked that the response example uses machine-readable names

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+
+: The percentage of gross ad revenue.
+
+- `GROSS_100`
+- `GROSS_85`
+- `GROSS_70`
+- `GROSS_60`
```
There are more options:

```suggestion
- `GROSS_100`
- `GROSS_95`
- `GROSS_85`
- `GROSS_80`
- `GROSS_70`
- `GROSS_65`
- `GROSS_60`
```


Also, today we [got](https://adjust.slack.com/archives/C031A94NE6N/p1730773179317829) a request to add 50

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+| `apps_network`                                                                                          | Returns a list of your apps and their associated store IDs.                                             |
+| `overview_metrics`                                                                                      | Returns a list of overview metrics.                                                                     |
+| `skad_metrics`                                                                                          | Returns a list of metrics relating to SKAdNetwork.                                                      |
+| `cohort_maturity`                                                                                       | Returns a list of metrics relating to the maturity of a cohort. Example: mature, immature.              |
```
There is a nice article about cohort_maturity, mb worth pinning it - https://help.adjust.com/en/article/how-cohorts-work?src=search_page#cohort-maturity

SlavaSkvortsov commented on src/content/docs/en/api/reports-service.mdoc:
```diff
@@ -0,0 +1,705 @@
+{% deflist %}
+`cohort_maturity`: `String`
+
+: Whether to include all cohort metric values (`immature`) or only mature cohorts `mature`. If you pass `mature`, any values for immature cohorts are displayed as zeros.
```
There is a nice article about cohort_maturity, mb worth pinning it - https://help.adjust.com/en/article/how-cohorts-work?src=search_page#cohort-maturity

Sporiff commented:
@SlavaSkvortsov @jmikovic I've updated the docs with the following changes:

1. I've incorporated links to documentation that you added here.
2. I've updated the types as pointed out for numerical comparisons.
3. I've updated the `GROSS_*` values.
4. I've removed the `filters_data` endpoint docs in favor of linking out to the glossaries on the HC.
5. I've swapped CSV and JSON endpoint docs so that CSV comes first.
6. I've removed the pivot report endpoint doc.
